Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm

```diff
@@ -616,6 +645,51 @@ static void runBasicTest(SchemeHandlerShouldBeAsync schemeHandlerShouldBeAsync)
     EXPECT_EQ(numberOfDecidePolicyCalls, 3);
 }
 
+TEST(ProcessSwap, SameSiteWindowOpenOpener)
+{
+    auto processPoolConfiguration = psonProcessPoolConfiguration();
+    processPoolConfiguration.get().processSwapsOnWindowOpenWithOpener = YES;
+    auto processPool = adoptNS([[WKProcessPool alloc] _initWithConfiguration:processPoolConfiguration.get()]);
+
+    auto webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    [webViewConfiguration setProcessPool:processPool.get()];
+    [webViewConfiguration preferences].javaScriptCanOpenWindowsAutomatically = YES;
+    auto handler = adoptNS([[PSONScheme alloc] init]);
+    [handler addMappingFromURLString:@"pson://www.webkit.org/main.html" toData:windowOpenWithNameCrossSiteOpenerTestBytes];
+    [handler addMappingFromURLString:@"pson://www.webkit.org/redirect.html" toData:windowOpenWithNameCrossSiteAfterRedirectOpenerTestBytes];
+
+    [handler addMappingFromURLString:@"pson://www.notwebkit.org/popup.html" toData:windowOpenWithNameCrossSiteAfterRedirectOpenerTestBytes];
+
+    [webViewConfiguration setURLSchemeHandler:handler.get() forURLScheme:@"PSON"];
+
+    auto webView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
+    auto navigationDelegate = adoptNS([[PSONNavigationDelegate alloc] init]);
+    [webView setNavigationDelegate:navigationDelegate.get()];
+    auto uiDelegate = adoptNS([[PSONUIDelegate alloc] initWithNavigationDelegate:navigationDelegate.get()]);
+    [webView setUIDelegate:uiDelegate.get()];
+
+    numberOfDecidePolicyCalls = 0;
+    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:@"pson://www.webkit.org/main.html"]];
+    [webView loadRequest:request];
+
+    TestWebKitAPI::Util::run(&done);
+    done = false;
+    TestWebKitAPI::Util::run(&didCreateWebView);
+    didCreateWebView = false;
+
+    TestWebKitAPI::Util::run(&done);
+
+    EXPECT_EQ(2, numberOfDecidePolicyCalls);
+
+    auto pid1 = [webView _webProcessIdentifier];
+    EXPECT_TRUE(!!pid1);
+    auto pid2 = [createdWebView _webProcessIdentifier];
+    EXPECT_TRUE(!!pid2);
+
+    EXPECT_NE(pid1, pid2);
+    TestWebKitAPI::Util::run(&didReceiveAlert);
```

Comment:
It might be worth using WKWebView._frames to inspect your frame tree.

Reply:
It seems the opener<->openee relationship doesn't change the results of WKWebView._frames both with this feature turned off and turned on.

Reply:
Why?  What would it take to make _frames give useful information for this test?